webview/WebView2_spec.md

```diff
@@ -0,0 +1,247 @@
+<br>
+<br>
+[Discussion of the core Microsoft Edge WebView2](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2)
+
+# Description
+
+The WebView is a control that allows for HTML content to be hosted within an application. It renders web content using the Microsoft Edge rendering engine.
+Use a WebView control to display richly formatted HTML content from a remote web server, dynamically generated code, or content files in your app. Rich content can also contain script code and communicate between the script and your app’s code.
+
+# Examples
+
+**NavigateToString:**
+The following code example demonstrates how to load local HTML into a WebView2 control.
+
+```
+myWebView2.NavigateToString("<html><body><h2>This is an HTML fragment</h2></body></html>”);
+```
+<br>
+
+**Source:**
+To set the initial content of the WebView2, set the Source property in XAML. The XAML parser automatically converts the string to a Uri.
+
+```
+<!-- Source file is on the web. -->
+<WebView2 x:Name="MyWebView_1" Source="http://www.contoso.com"/>
+
+<!-- Source file is in local storage. -->
+<WebView2 x:Name="MyWebView_2" Source="ms-appdata:///local/intro/welcome.html"/>
+
+<!-- Source file is in the app package. -->
+<WebView2 x:Name="MyWebView_3" Source="ms-appx-web:///help/about.html"/>
+
+<!-- no provided uri initialization -->
+<WebView2 x:Name="MyWebView_4"/>
+```
+<br>
+
+**ExecuteScriptAsync:**
+You can interact with the content of the WebView2 by using the ExecuteScriptAsync method to invoke or inject script into the WebView2 content, and the WebMessageReceived event to get information back from the WebView2 content.
+To invoke JavaScript inside the WebView2 content, use the ExecuteScriptAsync method. ExecuteScriptAsync returns a string which is the object result of the script execution converted to JSON and then returned as a string.
+
+```
+// You can use ExecuteScriptAsync to inject  //content into the web page.
+// Here, the text of a Xaml TextBox is written to a div in an HTML page hosted in //myWebView2.
+private async void Button_Click(object sender, RoutedEventArgs e)
+{
+    String functionString = 
+String.Format("document.getElementById('nameDiv').innerText = 'Hello, {0}';", nameTextBox.Text);
+await myWebView2.ExecuteScriptAsync(new string[] { functionString });
+}
+```
+
+<br>
+
+**WebMessageReceived:**
+Scripts in the WebView2 content can use window.chrome.webview.postMessage.postMessage(“Message”) with JSON parameters to send information back to your app. To receive these messages, handle the WebMessageReceived event.
+
+
+```
+// Event allows for notification when a webMessage event is fired from the JavaScript on the html side of the webview.
+
+myWebView2.WebMessageReceived += (WebView2 sender, WebView2WebMessageReceivedEventArgs args) =>
+{
+    // Important to validate that the Uri is what we expect from that webview.
```

Comment:
When would the event not come from the sender?

Reply:
It could be different if it is from an embedded <iframe>.